tools/check-abi.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+
+set -eux
+
+configure_and_build() {
+    cmake -S "$1" \
+        -DCMAKE_C_COMPILER=gcc -DCMAKE_BUILD_TYPE=None -DCMAKE_C_FLAGS="-g -Og -gdwarf-4" \
+        -DSECP256K1_ENABLE_MODULE_ECDH=ON \
+        -DSECP256K1_ENABLE_MODULE_RECOVERY=ON \
+        -DSECP256K1_ENABLE_MODULE_EXTRAKEYS=ON \
+        -DSECP256K1_ENABLE_MODULE_SCHNORRSIG=ON \
+        -DSECP256K1_ENABLE_MODULE_ELLSWIFT=ON \
```

Comment:
I bet that we'll forget to update this as soon as we add the next module. I suggest using dev mode instead. Yes, it's intended for developers, but I think it's acceptable in this case. Or what do you think?

Reply:
> I bet that we'll forget to update this as soon as we add the next module. I suggest using dev mode instead.

We might forget to update [`CMakePresets.json`](https://github.com/bitcoin-core/secp256k1/blob/master/CMakePresets.json) as well :D

For example, it lacks `SECP256K1_ENABLE_MODULE_ELLSWIFT`: https://github.com/bitcoin-core/secp256k1/blob/c545fdc374964424683d9dac31a828adedabe860/CMakePresets.json#L8-L12

Reply:
ellswift is enabled by default, so we don't need to enable it here explicitly.

Reply:
I mean, that "we'll forget to update this as soon as we add the next module" might be applied to the `CMakePresets.json` as well.

And using dev mode implies building examples, which is redundant for the ABI check.

Reply:
> I mean, that "we'll forget to update this as soon as we add the next module" might be applied to the `CMakePresets.json` as well.

Sure, but that doesn't make the point invalid. I think fewer code locations that we can forget to update is still better, even if it's not zero.  

> And using dev mode implies building examples, which is redundant for the ABI check.

Ok, what I had in mind is that it could still be disabled manually. But even if we don't, it doesn't really hurt to build the examples.

Reply:
> I suggest using dev mode instead.

Done.